### PR TITLE
Fix buckle in general

### LINKED ISSUE #1262: Entering a locker and buckling makes you teleport to 0,0 and enter noclip
Steps to reproduce: Spawn a locker and a chair next to it, enter the locker and close it, click the chair.

### LINKED ISSUE #1293: You can use clickdrag + chair buckle to cross over a table
https://streamable.com/wb5xel

### LINKED ISSUE #1304: Double clicking on a buckle-able object buckles you and instantly unbuckles you
Seems the unbuckle request is processed immediately after.

Videos:
[https://cdn.discordapp.com/attachments/675078881425752124/730164759424401468/2020-07-07_21-53-48.mp4](https://cdn.discordapp.com/attachments/675078881425752124/730164759424401468/2020-07-07_21-53-48.mp4)

[https://cdn.discordapp.com/attachments/675078881425752124/730164156623486986/2020-07-07_21-51-26.mp4](https://cdn.discordapp.com/attachments/675078881425752124/730164156623486986/2020-07-07_21-51-26.mp4)

### PR DESCRIPTION
**Pull request by DrSmugleaf**
Fixes #1262
Fixes #1293
Fixes #1304

- [x] Fixes being able to interact with entities while inside of a locker or any other container, when that entity is not the containing entity or an entity contained within the same container.

- [x] Removes free hand check when trying to buckle.

- [x] Fixes a buckled entity's sprite being drawn over the chair when facing up.

- [x] Fixes a character not standing up again after leaving a bed.

- [x] Fixes entering noclip when a locker is closed over a buckled entity and the chair is then deleted.

- [x] Adds a customizable buckle status effect icon depending on the strap.

- [x] Fixes not being unbuckled when entering a container if the strap is not put into the container as well.

- [x] Fixes being able to double click quickly to buckle and immediately unbuckle, teleporting through straps.

- [x] Reduces the buckling range to half of the regular interaction range.

- [x] Adds a range property to the buckle component.

- [x] Fixes being able to buckle through a table.

- [x] Fixes "You're already buckled in" message not appearing when clicking on a chair when already buckled.

- [x] Fixes buckle status icon texture error.

- [x] Adds a slight delay to unbuckling after having recently buckled (0.25 seconds).

- [x] Fixes "You can't reach there!" message being created twice.

- [x] Fixes not being able to buckle to entities in the same container.

- [x] ~Prevents Abraham from finding any new bugs.~

### CODE REVIEW

File: Content.Server/GameObjects/Components/Mobs/BuckleComponent.cs [hunk 1]
```diff
@@ -59,6 +60,22 @@ private void BuckleStatus()
             }
         }
 
+        private bool IsContained()
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Mobs/BuckleComponent.cs:63]: Unless the intent here is for this not to be recursive (for some reason) you can use `ContainerHelpers.IsInContainer(IEntity)` here.

File: Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs [hunk 2]
```diff
@@ -266,6 +267,16 @@ private void UserInteraction(IEntity player, GridCoordinates coordinates, Entity
...
+            // In a container where the attacked entity is not the container's owner and either the attacked entity is null, not contained or in a different container
+            if (ContainerHelpers.TryGetContainer(player, out var playerContainer) &&
+                attacked != playerContainer.Owner &&
+                (attacked == null ||
+                 !ContainerHelpers.TryGetContainer(attacked, out var attackedContainer) ||
+                 attackedContainer != playerContainer))
+            {
+                return;
+            }
+
```
Comment (Zumorica) [inline Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs:279]: My god this is ugly. Maybe make a few consecutive `if` statements instead? With comments, of course.

**Comment by DrSmugleaf** [comment]
Fixes #1304 1304

**Comment by ike709** [comment]
Fixes #1293 ?

**Pull request merged**